sdk/src/azure/core/az_span.c

```diff
@@ -382,13 +382,15 @@ AZ_NODISCARD az_result az_span_atod(az_span source, double* out_number)
 
   // Stack based string to allow thread-safe mutation.
   // The length is 8 to allow space for the null-terminating character.
+  // NOLINTNEXTLINE(readability-magic-numbers,  cppcoreguidelines-avoid-magic-numbers)
   char format[8] = "%00lf%n";
```

Comment:
why not adding the macro or enum for the 8 instead?

Reply:
I intentionally avoided adding named symbols for things that are one-off, or didn't have a useful name outside of the local instance. If desired, we can add them later.

In this case, it is the length of the format string. If one changes, we have to change the other. Putting a named constant far away hurts maintainability, so not worth the trade-off. Keeping these localized next to each other is valuable.